restful.module

```diff
@@ -388,6 +388,10 @@ function restful_get_restful_handler_for_path($path = NULL) {
   list($major_version, $minor_version) = \RestfulBase::getVersionFromRequest($path);
 
   $handlers[$path] = restful_get_restful_handler($resource, $major_version, $minor_version);
+  // Make sure we have a fresh request.
+  $request = restful_parse_request();
```

Comment:
Because the request comes cached and without filters at the point of calculating the count of results, I'll try to demonstrate it in a test